src/factories/box.ts

```diff
@@ -0,0 +1,34 @@
+import { differenceInMilliseconds, millisecondsInSecond } from 'date-fns'
+import { Graphics } from 'pixi.js'
+import { DEFAULT_TIME_COLUMN_SIZE_PIXELS } from '@/consts'
+import { RunGraphNode } from '@/models/RunGraph'
+import { waitForConfig } from '@/objects/config'
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function nodeBoxFactory() {
+  const config = await waitForConfig()
+  const box = new Graphics()
+
+  async function render(node: RunGraphNode): Promise<Graphics> {
+    const { background } = config.styles.node(node)
+
+    const right = node.start_time
+    const left = node.end_time ?? new Date()
+    const seconds = differenceInMilliseconds(left, right) / millisecondsInSecond
+    const boxWidth = seconds * DEFAULT_TIME_COLUMN_SIZE_PIXELS
+    const boxHeight = config.styles.nodeHeight - config.styles.nodeMargin * 2
```

Comment:
Worth calling out, this calculation used to depend on the x axis scale. But I realized last night it doesn't need to since we know how wide each second should be on the graph. Simplifies the dependencies here so it doesn't.